### DISCUSSION TITLE
Problem with FoxTelem receiving FSK on Fox in a box

### DISCUSSION THREAD

**Discussion by W7DGP** [alanbjohnston/CubeSatSim#289] [General]
While using FoxTelem on my laptop, I can receive and decode FSK. I get a good clean eye and no dropped data. Recently, I got the SD card from Amsat for Fox in a box and when I do the same thing with FoxTelem I get mostly dropped data with a very noisy eye. SNR goes from 9 to 2 and mostly averaging around 2.5. I'm using a Pi B+. I get the same with either Fun Cube Dongle Pro+ or RTL-SDR.com as the SDR.  
I believe signal strength is fine but the decoding is hit and miss. 
All other features in Fox in a box work well. APRS decodes and SSTV works the best I've seen. Before I show at another school, I would like to resolve this. Fox in a box is awesome except for this problem. Also, there is no decoding whatsoever running BPSK. That is a minor problem for me. Its FSK that will be used to demo the cubesat. 
I tried comparing the configs between my laptop (works great) and Fox in a box. They seem the same to me. Any help would be appreciated. Thanks, Larry W7DGP

**Comment by alanbjohnston** [comment]
Hi Larry,

Which Pi are you using? A Pi 3B+ or Pi 4? FoxTelem is marginal on a 3B+ although some people have luck with overclocking. 

While FoxTelem is running, you could open another window and type

top

This will tell you the CPU utilization. Control-C exits. 

73,
Alan

**Comment by W7DGP** [comment]
Thanks for the quick reply. I’m using a B+ but I tried a 4 with the same results….i think. My 4 is tethered to a touchscreen display that I’m having a hard time adjusting. . I also have a 5 (arrived yesterday) but no micro HDMI. My Amazon order was empty bag! Now I have to wait another week for the HDMI cable.
I am going to assume it’s the Pi and and get into my project boxes and focus on CPU strength.
Thanks again, Larry W7DGP

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by W7DGP** [comment]
Used top this morning. FoxTelem (java) uses 173% CPU. The Pi B+ is not going to work well for Fox in a box. I have a Pi 4 in a touchscreen enclosure that I plan to extricate and try. Also I may be able to get the Pi 5 going without having to plug it into a monitor. I can add the wifi stuff to the SD card then putty in and set it up for VNC. I have a lot of options.
73, Larry
________________________________

**Comment by W7DGP** [comment]
Today, I tried the Fox in a Box SD in my new Pi 5. Pi 5 needs an OS like Bookworm to work. This means the Fox in a Box will need to be built on that OS for PI 5 users. For me that will be a separate project. Next I will try a Pi 4. My objective is to get better performance when running FoxTelem on a Raspberry Pi Fox in a Box.
It was not a wasted effort. I did get the Pi 5 up and going for the first time with Bookworm. Its awesome and looks great on a 27 in monitor.
73, Larry W7DGP
________________________________

**Comment by W7DGP** [comment]
Today, I tried a Pi 4 (4GB) and it works great. I am using the AMSAT Fox in a Box SD card and its plug and play. Data looks great, clean eye and payloads adding up. CPU % varies from 90 to 100%.  My conclusion: Fox in a Box requires a Pi 4 for plug and play. Problem solved.
As to Pi 5, a good project would be to build a Pi 5 version using the Bookworm OS. There are very Pi 5s out there at this time so it would be a good project for 2024.

**Comment by alanbjohnston** [comment]
I'm working on a Bookworm version of the AMSAT Fox-in-a-Box image, which will be FIABv4.  I have it mostly working, but Java in Bookworm doesn't seem very stable, and FoxTelem gives some errors.  So for now, I would stick with a Pi 4. If you need a Bookworm version, let me know and I can share the work-in-progress, but it is not ready for prime time yet.

Alan
